ball_detection.py

```diff
@@ -1,43 +1,85 @@
 import cv2
 import numpy as np
-import imutils
+
+# known diamter of ball from 90in way (not accounting for angle)
+KNOWN_DIAMETER_PX = 156  # px
+# known distance from cam to closest ball (7.5', not accounting for angle)
+KNOWN_DISTANCE = 90  # in
+# known diameter of ball
+KNOWN_DIAMETER_IN = 7  # in
+# focal len = 90 * 156 / 7
+FOCAL_LENGTH = 2005.714286  # px
 
 RADIUS_THRESH = 25  # modify
 KNOWN_RADIUS = 3.5  # in
+RADIUS_THRESH = 10  # modify
 
 
-def detectBall(frame):
-    frame = cv2.GaussianBlur(frame, (5, 5), 0)
-    hsv_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
-    lower_yellow = np.array([22, 28, 82])  # values from field image #fix this range
+def distance(flength, kwidth, pwidth):
+    dist = ((kwidth * flength) / pwidth) / 12
+    print("DISTANCE: ", dist)
+    return round(dist, 1)
+
+
+def detectBall(frame, w, h):
+    x_coords = []
+    dist = 0
+    output = frame.copy()
+    lower_yellow = np.array([17, 130, 130])  # values from field image; fix this range
     higher_yellow = np.array([30, 255, 255])  # values from field image
 
+    hsv_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
+    hsv_frame = cv2.GaussianBlur(hsv_frame, (5, 5), 0)
     mask = cv2.inRange(hsv_frame, lower_yellow, higher_yellow)
-    mask = cv2.erode(mask, None, iterations=2)
-    mask = cv2.dilate(mask, None, iterations=2)
+    mask = cv2.erode(mask, None, iterations=7)
+    mask = cv2.dilate(mask, None, iterations=10)
+
+    mask = mask[145:h, 0:w]
+    output = output[145:h, 0:w]
 
     cnts = cv2.findContours(mask.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
-    cnts = imutils.grab_contours(cnts)
+    cnts = grab_contours(cnts)
 
     if len(cnts) > 0:
         max_cnt = max(cnts, key=cv2.contourArea)
         for cnt in cnts:
             ((x, y), radius) = cv2.minEnclosingCircle(cnt)
+            print(radius)
 
             # FOR DETERMINING DISTANCE TO CLOSEST BALL:
-            # if any(cnt[0][0] == max_cnt[0][0]):
-            #     dist = distance(focal_length, known_radius, radius)
-            #     cv2.putText(frame, '{} in'.format(str(dist)), (100, 100), cv2.FONT_HERSHEY_SIMPLEX, 2, (0, 0, 255), 2)
+            if any(cnt[0][0] == max_cnt[0][0]):
```

Comment:
What's the purpose of this check? max_cnt is selected from within cnt, so when would this be false?